src/nvim/regexp_nfa.c

```diff
@@ -4781,6 +4781,7 @@ static long find_match_text(colnr_T startcol, int regstart, char_u *match_text)
     if (match
         /* check that no composing char follows */
         && !(enc_utf8
+             && STRLEN(regline) > (size_t)(col + len2)
```

Comment:
Same as above, `regline` is a global variable and there are some function calls in the loop. Unless you inspected every possible code path in the block, it is safer to just leave it there.